luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/SingleDataSourcePhysicalTableFactory.java

```diff
@@ -0,0 +1,94 @@
+    private static final String ENTITY_TYPE = "single data source physical table";
+
+    /**
+     * Build the parameter for the subclass of SingleDataSourceParams to use.
+     *
+     * @param name  name of the LuthierTable as a String
+     * @param configTable  ObjectNode that points to the value of corresponding table entry in config file
+     * @param resourceFactories  should be the industrial park that needs to be paseed along to the builders
+     *
+     * @return  a param bean that contains information need to build the PhysicalTable
+     */
+    LuthierPhysicalTableParams buildParams(
+            String name,
+            ObjectNode configTable,
+            LuthierIndustrialPark resourceFactories
+    ) {
+        validateFields(name, configTable);
+
+        LuthierPhysicalTableParams params = new LuthierPhysicalTableParams();
+        params.tableName = new LuthierTableName(name);
+        params.timeGrain = DefaultTimeGrain
+                .valueOf(configTable.get("granularity").textValue().toUpperCase(Locale.US))
+                .buildZonedTimeGrain(DateTimeZone.forID(configTable.get("dateTimeZone").textValue()));
+        params.columns = new LinkedHashSet<>();
+
+        JsonNode dimensionsNode = configTable.get("dimensions");
+        dimensionsNode.forEach(
+                node -> params.columns.add(new DimensionColumn(resourceFactories.getDimension(node.textValue())))
+        );
+
+        JsonNode metricsNode = configTable.get("metrics");
+        metricsNode.forEach(
+                node -> params.columns.add(new MetricColumn(node.textValue()))
+        );
+
+        params.logicalToPhysicalColumnNames = new LinkedHashMap<>();
+        configTable.get("logicalToPhysicalColumnNames").forEach(
+                node -> params.logicalToPhysicalColumnNames.put(
+                        node.get("logicalName").textValue(),
+                        node.get("physicalName").textValue()
+                )
+        );
+        params.metadataService = resourceFactories.getMetadataService();
+        return params;
+    }
+
+    /**
+     * Helper function to validate only the fields needed in the parameter build.
+     *
+     * @param name  name of the LuthierTable as a String
+     * @param configTable  ObjectNode that points to the value of corresponding table entry in config file
+     */
+    protected void validateFields(String name, ObjectNode configTable) {
+        LuthierValidationUtils.validateField(configTable.get("granularity"), ENTITY_TYPE, name, "granularity");
+        LuthierValidationUtils.validateField(configTable.get("dateTimeZone"), ENTITY_TYPE, name, "dateTimeZone");
+        LuthierValidationUtils.validateField(configTable.get("dimensions"), ENTITY_TYPE, name, "dimensions");
+        LuthierValidationUtils.validateField(
+                configTable.get("logicalToPhysicalColumnNames"),
+                ENTITY_TYPE,
+                name,
+                "searchProvider"
```

Comment:
Field name here is wrong. It should `"logicalToPhysicalColumnNames"` not `"searchProvider"`.

Reply:
😬Copy and Paste fail